.github/workflows/frontend.yml

```diff
@@ -70,49 +71,41 @@ jobs:
           echo "::add-matcher::.github/tsc.json"
           echo "::add-matcher::.github/eslint-stylish.json"
 
-      - name: eslint logic
-        id: eslint
-        if: (github.ref == 'refs/heads/master' || needs.files-changed.outputs.eslint_config == 'true' || needs.files-changed.outputs.yarn_lockfile == 'true')
-        run: echo "all-files=true" >> "$GITHUB_OUTPUT"
-
-      # Lint entire frontend if:
-      # - this is on main branch
-      # - eslint configuration in repo has changed
-      # - yarn lockfile has changed (i.e. we bump our eslint config)
-      - name: eslint
-        if: steps.eslint.outputs.all-files == 'true'
-        env:
-          # Run relax config on main branch (and stricter config for changed files)
-          SENTRY_ESLINT_RELAXED: 1
-        run: yarn lint
+      - name: lint
+        if: github.ref == 'refs/heads/master' || needs.files-changed.outputs.lintable_rules_changed != 'true' || needs.files-changed.outputs.lintable_css_in_js_rules_changed != 'true'
+        run: |
+          yarn lint:biome
+          yarn lint:prettier
+          yarn lint:js
+          yarn lint:css
+
+      - name: biome
+        if: github.ref != 'refs/heads/master'
+        run: yarn fix:biome
+
+      - name: prettier (changed files only)
+        if: github.ref != 'refs/heads/master' && needs.files-changed.outputs.lintable_rules_changed != 'true'
+        run: yarn prettier --write ${{ needs.files-changed.outputs.lintable_modified_files }}
```

Comment:
Prettier (for changed files) was completely missing from the before file.  It was only being run in prod, which means that js files could get merged and then cause master branch to break. It' happened a few times last week :(

Reply:
Ah, previously iirc we were using eslint/prettier plugin to handle formatting, so this was probably missed when we added prettier back

Reply:
ya, that's probably what happened. feels like too many tools in play right now tbh, i think i would like to revert and go back to the simpler setup and only have eslint coordinate everything. that'll mean fewer custom config files in all the repo's too.